pkg/operator/controller/controller_service_monitor.go

```diff
@@ -43,6 +46,10 @@ func serviceMonitorName(ci *ingressv1alpha1.ClusterIngress) types.NamespacedName
 }
 
 func desiredServiceMonitor(ci *ingressv1alpha1.ClusterIngress, svc *corev1.Service, deployment *appsv1.Deployment) *unstructured.Unstructured {
+	if svc == nil || deployment == nil {
+		return nil
+	}
```

Comment:
I don't think you need to check `svc == nil || deployment == nil` both here and in `ensureServiceMonitor`.

Reply:
Same reason as above.